### ISSUE TITLE
Rewrite engine renderer to lag less.

### ISSUE THREAD

**Issue by asiekierka** [BuildCraft/BuildCraft#3042]
Right now the entire Engine is a TESR.

It needs to be split into an ISBRH and a TESR, with severe display list (or even VBO) based caching for the moving part due to the fact engines are used in large arrays.

**Comment by asiekierka** [comment]
Right now the Engine is split into an ISBRH and a TESR! This is better, but not quite there yet.

**Comment by AlexIIL** [comment]
Actually we need to remove the block model part so it's only a TESR. With fry's new fastTesr class it should speed things up but it has a tonne of bugs with using both.

**Comment by asiekierka** [comment]
That's a bad idea IMHO. FastTESRs still have a massive load compared to ISBRHs, it's just smaller than regular TESRs.

I'd rather wait until FastTESRs are fixed up. :)

**Comment by AlexIIL** [comment]
This isn't opinion, it's comparing to when I split them up. We need to use a TESR at all times anyway (changing the block model a lot is a very bad idea).

**Comment by asiekierka** [comment]
Yes, but then we send more data a frame to the graphics card. This might cause issues.

**Comment by AlexIIL** [comment]
If anything is being a render issue its the quarry. A single one of them is chewing up about 5% of my render time. A single engine is negligable.

**Comment by asiekierka** [comment]
I've seen 20-30 engines take up quite a lot of rendering time on BC 6.4 on
chilm's machine, IIRC.

Also, I *think* adding display list support to RenderBoxProvider might fix
this one to an extent. That, or maybe it's the mechanical arm?

2016-03-12 15:04 GMT+01:00 AlexIIL <notifications@github.com>:

**Comment by AlexIIL** [comment]
Oh right, people generally have more engines than quarrys. derp.

**Comment by AEnterprise** [comment]
is this still relevant?

**Comment by AlexIIL** [comment]
Yes.

Most likely happening in BC8.
